Minor prerelease updates (#225)
WPTRT/WPThemeReview: CHANGELOG.md

  You can check the documentation of each of the functions used in determining the template hierarchy: [get_category_template()](https://developer.wordpress.org/reference/functions/get_category_template), [get_author_template()](https://developer.wordpress.org/reference/functions/get_author_template), [get_page_template()](https://developer.wordpress.org/reference/functions/get_page_template), [get_tag_template()](https://developer.wordpress.org/reference/functions/get_tag_template).
  Or you can check the in depth documentation about the [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy).
- New `WPThemeReview.Privacy.ShortenedURLs` sniff: detects the usage of shortened URLs. [Handbook rule](https://make.wordpress.org/themes/handbook/review/required/#privacy).
- New `WPThemeReview.CoreFunctionality.PostsPerPage` sniff: adds a warning if `-1` is used in `posts_per_page` setting while querying posts, due to detrimental effects it has on the query speed.
- The `WordPress.PHP.IniSet` rule was added to the ruleset about setting ini configuration during runtime.

> - The `WordPress.WP.DeprecatedParameterValues` rule was added to the ruleset about usage of deprecated parameter values in WP functions. The sniff will provide alternative based on the parameter passed.

Reviewer: ```suggestion
- The `WordPress.WP.DeprecatedParameterValues` rule was added to the ruleset about usage of deprecated parameter values in WP functions. The sniff will suggest an alternative based on the parameter passed.
```

  The new WPTRCS native version of the sniff overloads the prefix check for variables only and will bow out if the file being scanned has a typical theme template file name.
  For all other files, it will fall through to the WPCS native sniff.
  For all other constructs in theme template files, i.e. namespaces, classes, functions etc, the checks from the WPCS native sniff will still be run.
  Notes:
    * The new sniff adds a public `$allowed_folders` property to whitelist files in specific folders of a theme as template files.
    The `ruleset.xml` file sets this property to a limited set of folders whitelisted by default.
    The default property value can be overruled and/or extended from a custom ruleset.
    * Similar to the WPCS FileNameSniff, this sniff does not currently allow for mimetype sublevel only theme template file names, such as `plain.php`.

### Changed

- Travis updates:
  - Run the code style related and ruleset checks in separate stages.
  - Removed `sudo: false` setting.
  - Test agains high/low WPCS versions due to sniffs that are extending the WPCS native sniffs.

> - Update forbidden functionality with cron functions in the `WPThemeReview.PluginTerritory.ForbiddenFunctions` sniff.

Reviewer: While this is a change to the sniff, it results in new checks being added to TRTCS, so IMO, this entry should be in the `Added` section.

- Added XSD schema tags and validated the ruleset against schema (PHPCS 3.2+/3.3.2+).

> - Update forbidden functionality with (un)register editor blocks functions in the `WPThemeReview.PluginTerritory.ForbiddenFunctions` sniff.

Reviewer: Same remark as for the cron functions.
These two entries could possibly be combined into one.

- Updated the WPCS to 2.1.0 version.
- Updated the suggested installer version.
- Composer tweaks: improve readability of script section
- Ruleset tweaks: limit PHPCompatibility to PHP files

> - Update the WPCS namespace to `WordPressCS`

Reviewer: ```suggestion
- Updated the WPThemeReview codebase, where relevant, for compatibility with WPCS 2.0+.
```

### Fixed

- Removal of HTML from error message about adding menu pages in `WPThemeReview.PluginTerritory.NoAddAdminPages` sniff.

> - Minor grammar changed in ruleset.

Reviewer: ```suggestion
- Minor grammar changes in the ruleset.
```
